drivers/slipdev/slipdev.c

```diff
@@ -218,58 +245,27 @@ static int _recv(netdev_t *netdev, void *buf, size_t len, void *info)
     if (buf == NULL) {
         if (len > 0) {
             /* remove data */
-            for (; len > 0; len--) {
-                int byte = tsrb_get_one(&dev->inbuf);
-                if ((byte == (int)SLIPDEV_END) || (byte < 0)) {
-                    /* end early if end of packet or ringbuffer is reached;
-                     * len might be larger than the actual packet */
-                    break;
-                }
-            }
+            crb_consume_chunk(&dev->rb, NULL, len);
         } else {
             /* the user was warned not to use a buffer size > `INT_MAX` ;-) */
-            res = (int)tsrb_avail(&dev->inbuf);
+            crb_get_chunk_size(&dev->rb, (size_t*)&res);
```

Comment:
For us I assume it is true, but in general it is not safe to assume `sizeof(int) == sizeof(size_t)`.
So writing a `size_t` to an `int` causes trouble on at least 64bit Linux. For  cleanness i would prefer an intermediate `size_t` variable.

Reply:
There is work on getting `native` running on 64 bit. Also a 64 bit RISC-V port is in the works (stalled currently). So I think we should already prepare the code to be 64 bit safe.